Cython/Compiler/Nodes.py

```diff
@@ -2411,11 +2424,16 @@ def analyse_declarations(self, env):
 
         type.is_const_method = self.is_const_method
         type.is_static_method = self.is_static_method
+
         self.entry = env.declare_cfunction(
             name, type, self.pos,
             cname=cname, visibility=self.visibility, api=self.api,
             defining=self.body is not None, modifiers=self.modifiers,
             overridable=self.overridable)
+        if is_property:
+            self.entry.is_property = 1
+            env.property_entries.append(self.entry)
+            env.cfunc_entries.remove(self.entry)
```

Comment:
This makes me wonder why we shouldn't be calling `declare_property()` here in the first place, instead of `declare_cfunction()`. What do you think? It seems to do the right thing, pretty much.

Reply:
`CClassScope.declare_property` does not dive into the wrapped function the way `CClassScope.declare_cfuntion` does. I played with this for a while but could not get it to work cleanly, without in the end calling `declare_cfunction` anyway.

Reply:
done, extra test added. Since I want to print the `type` of the unhandled instance, I needed to rename `type` -> `type` in that function.